test/bundler/bundler_edgecase.test.ts

```diff
@@ -1443,6 +1443,32 @@ describe("bundler", () => {
       "/entry.ts": [`"Y" has already been declared`],
     },
   });
+  // This specifically only happens with 'export { ... } from ...' syntax
+  itBundled("edgecase/EsmSideEffectsFalseWithSideEffectsExportFrom", {
+    files: {
+      "/file1.js": `
+        import("./file2.js");
```

Comment:
Tests for:
- `require` but the same as this
- One with code splitting enabled where two entry points import the same other files